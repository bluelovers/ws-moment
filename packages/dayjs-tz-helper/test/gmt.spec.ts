//@noUnusedParameters:false
/// <reference types="jest" />
/// <reference types="node" />
/// <reference types="expect" />

import { basename, extname } from 'path';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import tzDayjsSafeParse from '../src';

beforeAll(async () =>
{
	dayjs.extend(utc);
	dayjs.extend(timezone);
});

describe(`GMT`, () =>
{
	beforeAll(async () =>
	{
		dayjs.tz.setDefault('GMT');
	});

	test.skip(`dummy`, () => {});

	test.each([
		'Fri, 31 Mar 2023 04:00:00 GMT',
		'Fri, 31 Mar 2023 04:00:00',
	])(`%s`, (input) =>
	{

		let actual = tzDayjsSafeParse(input, 'GMT');
		let expected;

		expect({
			input,
			actual,
			toISOString: actual.toISOString(),
			dayjs: dayjs(input),
			[`dayjs:utc`]: dayjs.utc(input),
			[`dayjs:tz`]: dayjs.tz(input, 'GMT'),
		}).toMatchSnapshot({
			toISOString: '2023-03-31T04:00:00.000Z',
		});

	});

})
