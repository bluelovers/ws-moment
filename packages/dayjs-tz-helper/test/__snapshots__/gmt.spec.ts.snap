// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GMT Fri, 31 Mar 2023 04:00:00 1`] = `
{
  "actual": "2023-03-31T04:00:00.000Z",
  "dayjs": "2023-03-30T20:00:00.000Z",
  "dayjs:tz": "2023-03-30T20:00:00.000Z",
  "dayjs:utc": "2023-03-30T20:00:00.000Z",
  "input": "Fri, 31 Mar 2023 04:00:00",
  "toISOString": "2023-03-31T04:00:00.000Z",
}
`;

exports[`GMT Fri, 31 Mar 2023 04:00:00 GMT 1`] = `
{
  "actual": "2023-03-31T04:00:00.000Z",
  "dayjs": "2023-03-31T04:00:00.000Z",
  "dayjs:tz": "2023-03-31T04:00:00.000Z",
  "dayjs:utc": "2023-03-31T04:00:00.000Z",
  "input": "Fri, 31 Mar 2023 04:00:00 GMT",
  "toISOString": "2023-03-31T04:00:00.000Z",
}
`;
